autoload/lsp/ui/vim/utils.vim

```diff
@@ -176,12 +178,21 @@ function! s:is_file_uri(uri) abort
     return stridx(a:uri, 'file:///') == 0
 endfunction
 
-function! s:get_symbol_text_from_kind(kind) abort
-    return has_key(s:symbol_kinds, a:kind) ? s:symbol_kinds[a:kind] : 'unknown symbol ' . a:kind
+function! s:get_symbol_text_from_kind(server, kind) abort
+    if !has_key(s:symbol_kinds, a:server)
+        let l:server_info = lsp#get_server_info(a:server)
+        if has_key (l:server_info, 'config') && has_key(l:server_info['config'], 'symbol_kinds')
+            let s:symbol_kinds[a:server] = extend(s:default_symbol_kinds, l:server_info['config']['symbol_kinds'])
+        else
+            let s:symbol_kinds[a:server] = s:default_symbol_kinds
+        endif
+    endif
+        let l:symbol_kinds = s:symbol_kinds[a:server]
+    return has_key(l:symbol_kinds, a:kind) ? l:symbol_kinds[a:kind] : 'unknown symbol ' . a:kind
 endfunction
 
 function! lsp#ui#vim#utils#get_symbol_kinds() abort
```

Comment:
I thought about that, but the `valueSet` contains only the numbers, not the text, so it shouldn't make a difference (it could be longer, but that would be against the specification, so the server can't be expected to deal with it?

Reply:
I was thinking about adding support for additional kinds, but you're right, that's against the specification. Does texlab have additional kinds, or does it just use the predefined ones?

Reply:
It's not illegal to extend the specification. But texlab sticks to the number in the current specification (26 resp. 25). 

It's possible to do this, but requires some refactoring (making the s:symbol_kinds a global variable that is constructed during registration); to quote Bartleby: I'd rather not. (I.e., leave it for a future PR when someone actually requests this for a specific server.)

Reply:
You should be able to reuse the code in `lsp#omni#get_kind_text`, but I'm fine with leaving it as-is.

Reply:
Yes, that's what I meant by "refactor" -- there's no new code necessary, but the current code has to be rearranged somewhat (and exposed in the case of `symbolKind`). Let's leave it for later, then.